### PR TITLE
Debugger tests fixes/improvements

### PR DESCRIPTION
**Pull request by eduser25**
- Removed fakes, added mocks for pkg/debugger interface types
- Moved available tests to use mocks
- Added tests for debugger/server and debugger/certificate
- Created/rebased tests to use `testing` instead of `Ginko`

`7.7%` -> `36.0%`

Please describe the motivation for this PR and provide enough
information so that others can review it.

- Tests / CI System      [X]

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

**Comment by eduser25** [comment]
Will rebase w/o using Ginko

**Review by ksubrmnn — APPROVED** [review]

**Review by snehachhabria — DISMISSED** [review]

File: pkg/debugger/policy_test.go [hunk 1]
```diff
@@ -1,71 +1,38 @@
 package debugger
 
 import (
-	"fmt"
 	"net/http/httptest"
 	"testing"
-	"time"
 
+	gomock "github.com/golang/mock/gomock"
```
Comment (shashankram) [inline pkg/debugger/policy_test.go:7]: we don't need an import alias here.

**Review by shashankram — APPROVED** [review]

**Review by snehachhabria — APPROVED** [review]

File: pkg/debugger/policy_test.go (hunk 1 above)
Comment (eduser25) [reply pkg/debugger/policy_test.go:7]: fixing

Comment (shashankram) [reply pkg/debugger/policy_test.go:7]: Don't worry about it, can be handled later. We have other occurrences of this.

**Review by ksubrmnn — APPROVED** [review]

**Pull request merged**